### PR TITLE
fix restricted roles becoming gang leaders

### LINKED ISSUE #62384: Captain can roll "family head" antagonist (and so can other antags - traitors, wizards, changelings)
## Reproduction:

image

### PR DESCRIPTION
**Pull request by phasenoisepon**
## About The Pull Request

closes https://github.com/tgstation/tgstation/issues/62384

## Changelog

:cl:
fix: restricted roles cannot become gang leaders anymore
/:cl:

### CODE REVIEW

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -269,8 +269,11 @@
...
 			candidates -= player
-		else if(player.mind && (player.mind.special_role || player.mind.antag_datums?.len > 0))
-			candidates -= player
+		else if(player.mind)
+			if(player.mind.assigned_role.title in restricted_roles)
+				candidates -= player
+			else if(player.mind.special_role || player.mind.antag_datums?.len > 0)
+				candidates -= player
 		else if(HAS_TRAIT(player, TRAIT_MINDSHIELD))
 			candidates -= player
```
Comment (ShizCalev) [inline code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:278]: small bug here.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm (hunk 1 above)
Comment (phasenoisepon) [reply code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:278]: Do tell?

Comment (ShizCalev) [reply code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:278]: i'm a little rusty, but won't we never reach this point if the player.mind check passes?

Comment (phasenoisepon) [reply code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:278]: that would be embarrassing, you're correct

**Comment by Seris02** [comment]
see my comment on #62384

**Pull request closed without merging**